lib/sortChannels.ts

```diff
@@ -1,13 +1,55 @@
-import { ChannelStats } from "@/types/ChannelStats";
+import { ChannelStats, RankCriterion } from "@/types/ChannelStats";
 
 function sortChannels(channelStats: ChannelStats[]) {
-  return channelStats.sort((a, b) => {
-    return (
-      (b.numberOfNotes! - a.numberOfNotes!) ||
-      (b.numberOfSongs - a.numberOfSongs) ||
-      (b.numberOfCurators - a.numberOfCurators)
-    );
+  const criteria: RankCriterion[] = [
+    { name: 'numberOfNotes', weight: .6 },
+    { name: 'numberOfSongs', weight: .2 },
+    { name: 'numberOfCurators', weight: .2 },
+  ];
+
+  const normalizedData = normalizeData(channelStats, criteria);
+  const weightedScores = calculateWeightedScores(normalizedData, criteria);
+  const rankedItems = rankItems(channelStats, weightedScores);
+
+  return rankedItems;
+}
+
+function normalizeData(data: ChannelStats[], criteria: RankCriterion[]) {
```

Comment:
each file should do EXACTLY one thing.
- please create a new lib for `normalizeData`